sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsLoggerConfig.java

```diff
@@ -0,0 +1,101 @@
+ * for an operation or not
+ */
+public final class CosmosDiagnosticsLoggerConfig {
+    private Duration pointOperationLatencyThreshold;
+    private Duration feedOperationLatencyThreshold;
+    private float requestChargeThreshold;
+
+    /**
+     * Creates an instance of the CosmosDiagnosticsLoggerConfig class with default parameters
+     */
+    public CosmosDiagnosticsLoggerConfig() {
+        this(Duration.ofSeconds(1), Duration.ofSeconds(3), 1000);
+    }
+
+    /**
+     * Creates an instance of the CosmosDiagnosticsLoggerConfig class with custom parameters
+     * @param pointOperationLatencyThreshold the latency threshold for all point operations
+     * @param feedOperationLatencyThreshold the latency threshold for feed operations (query and change feed - and bulk)
+     * @param requestChargeThreshold the request charge threshold
+     */
+    public CosmosDiagnosticsLoggerConfig(
+        Duration pointOperationLatencyThreshold,
+        Duration feedOperationLatencyThreshold,
+        float requestChargeThreshold) {
+
+        this.pointOperationLatencyThreshold = pointOperationLatencyThreshold;
+        this.feedOperationLatencyThreshold = feedOperationLatencyThreshold;
+        this.requestChargeThreshold = requestChargeThreshold;
+    }
+
+    /**
+     * Diagnostics for point operations will be logged if their end-to-end latency exceed this value
+     * @return the end-to-end latency threshold for point operations
+     */
+    public Duration getPointOperationLatencyThreshold() {
+        return this.pointOperationLatencyThreshold;
+    }
+
+    /**
+     * Diagnostics for point operations will be logged if their end-to-end latency exceed this value
+     * @param newPointOperationLatencyThreshold The end-to-end latency threshold for point operations
+     * @return the CosmosDiagnosticsLoggerConfig instance
+     */
+    public CosmosDiagnosticsLoggerConfig setPointOperationLatencyThreshold(
+        Duration newPointOperationLatencyThreshold) {
+
+        this.pointOperationLatencyThreshold = newPointOperationLatencyThreshold;
+
+        return this;
+    }
+
+    /**
+     * Diagnostics for feed operations will be logged if their end-to-end latency exceed this value
+     * @return the end-to-end latency threshold for feed operations
+     */
+    public Duration getFeedOperationLatencyThreshold() {
+        return this.feedOperationLatencyThreshold;
+    }
+
+    /**
+     * Diagnostics for feed operations will be logged if their end-to-end latency exceed this value
+     * @param newFeedOperationLatencyThreshold The end-to-end latency threshold for feed operations
+     * @return the CosmosDiagnosticsLoggerConfig instance
+     */
+    public CosmosDiagnosticsLoggerConfig setFeedOperationLatencyThreshold(
+        Duration newFeedOperationLatencyThreshold) {
+
+        this.feedOperationLatencyThreshold = newFeedOperationLatencyThreshold;
+
+        return this;
+    }
```

Comment:
Just curious - is invoking a stored procedure considered a point operation or feed operation? May be it is a point operation even if it updates multiple records...

Reply:
No, SP is not a point operation - it acts on entire logical partition - not a single document